responses/registries.py

```diff
@@ -67,3 +67,25 @@ def replace(self, response: "BaseResponse") -> None:
                 "Response is not registered for URL {}".format(response.url)
             )
         self.registered[index] = response
+
+
+class OrderedRegistry(FirstMatchRegistry):
+    def find(
+        self, request: "PreparedRequest"
+    ) -> Tuple[Optional["BaseResponse"], List[str]]:
+
+        if not self.registered:
+            return None, ["No more registered responses"]
+
+        response = self.registered.pop(0)
+        match_result, reason = response.matches(request)
+        if not match_result:
+            self.reset()
```

Comment:
Won't wiping the registry remove other mocks that haven't been called yet?

Reply:
@markstory 
yes, it will
the point is that if we want to modify error message without changing existing code, then we need to clean up all registered `responses` and fill only existing one